app/controllers/bookings_controller.rb

```diff
@@ -9,7 +9,7 @@ def create
     @booking.vehicle = Vehicle.find(params[:vehicle_id])
     @booking.user = current_user
     authorize @booking
-    @booking.save ? (redirect_to user_path(current_user)) : (render 'vehicles/show')
+    @booking.save ? (redirect_to user_path(current_user)) : (redirect_to vehicle_path(@booking.vehicle))
```

Comment:
do you really want to redirect? i think usually you don't do that because then the page will reload and the input data is gone...

Reply:
That's true, but this only happens if somehow the user has managed to select invalid dates, in which case they'll have to select again anyway. The previous version was broken because it couldn't go back to the correct vehicle page, but now all that happens is the data selection is cleared

Reply:
Ah and also (this is not too relevant since this is a small application, but whatever) a booking might be rejected on the backend if someone else has just created an overlapping booking, but without a page refresh the date picker can't know that those dates have become unavailable